docs/tracing.md

```diff
@@ -0,0 +1,74 @@
+# Tracing
+
+## Introduction
+
+Instrumentation based tracing is implemented with
+[`log`](https://github.com/rust-lang/log) and
+[`log_instrument`](https://github.com/JonathanWoollett-Light/log-instrument),
+outputting a `Trace` level log when entering and exiting every function.
+
+It is disabled by default at compile-time, as such the
+**tracing functionality has no impact on the release binary.**
```

Comment:
nit: 
```suggestion
It is disabled by default at compile-time, meaning the
tracing functionality has no runtime or memory impact on the release binary.
```

Reply:
Mh, in this case maybe just say "performance", as that can refer to both execution time and memory?

Reply:
If we specify `performance` we are saying it may have impacts other than performance. E.g. it might change a log message.

Reply:
That makes sense :)